### PR TITLE
beforeScreenshot

### PR DESCRIPTION
**Pull request by tavyandy97**
Changelog: 

- Added a beforeScreenshot argument which shall be called just before the screenshot is taken and provides access to puppeteer's page element

### CODE REVIEW

**Comment by tavyandy97** [comment]
Pull request for #61

File: src/screenshot.js [hunk 1]
```diff
@@ -22,6 +23,7 @@ module.exports = {
     }
     await page.setContent(html, { waitUntil })
     const element = await page.$('body')
+    await beforeScreenshot(page)
```
Comment (frinyvonnick) [inline src/screenshot.js:26]: Could you add a test in `screenshot.spec.js` that ensures this feature works?

File: src/screenshot.js [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 module.exports = {
-  makeScreenshot: async function(page, {
+  makeScreenshot: async function (page, {
     output,
     type,
     quality,
     encoding,
     content,
     html,
+    beforeScreenshot = () => { },
```
Comment (frinyvonnick) [inline src/screenshot.js:11]: This line set a default value which works fine only is nothing is passed to the function. If anything else is passed to this argument it won't works (string, numbers, null, ...). I think we should check if beforeScreenshot is a `function` before calling it. What do you think of it?

**Review by frinyvonnick — APPROVED** [review]
Thank you @tavyandy97 for opening a pull request so fast 👏 

I left two comments. If you take them in account it looks good to me 👌

File: src/screenshot.js (hunk 2 above)
Comment (tavyandy97) [reply src/screenshot.js:11]: I agree and will implement a typecheck for `beforeScreenshot`

File: src/screenshot.js (hunk 1 above)
Comment (tavyandy97) [reply src/screenshot.js:26]: Will do 👍

**Comment by tavyandy97** [comment]
Hey @frinyvonnick! Made the changes you suggested.

**Comment by frinyvonnick** [comment]
Nice job 👍 

I'll publish it as soon as possible 🚀

**Comment by frinyvonnick** [comment]
It has been published in [3.1.0](https://github.com/frinyvonnick/node-html-to-image/releases/tag/v3.1.0) 🚀

**Pull request merged**